### ISSUE TITLE
Using #[schema(title = ...)] does not work for new type / tuple structs of type Vec<T>

### ISSUE THREAD

**Issue by pdavid31** [juhaku/utoipa#850]
Let me start of by saying: Thank you for your great work on this crate.

Something that I noticed is that using `#[schema(title = "...")]` does not seem to have an effect on tuple structs containing vectors.
I created a minimal example to reproduce the issue.

```toml
[package]
name = "utoipa-test"
version = "0.1.0"
edition = "2021"

[dependencies]
actix-web = "4.4.1"
utoipa = "4.2.0"
utoipa-swagger-ui = { version = "6.0.0", features = ["actix-web"] }
```

```rust
use std::{io::Error, net::Ipv4Addr};

use actix_web::{App, HttpServer};
use utoipa::{OpenApi, ToSchema};
use utoipa_swagger_ui::SwaggerUi;

#[derive(ToSchema)]
#[schema(title = "This is an integer using the new type idiom / tuple struct")]
pub struct NewTypeInteger(Vec<i64>);

#[derive(ToSchema)]
#[schema(title = "This is an integer using a regular struct")]
pub struct NamedFieldInteger {
    i: Vec<i64>,
}

#[derive(OpenApi)]
#[openapi(
    info(title = "Minimal example",),
    components(schemas(NewTypeInteger, NamedFieldInteger))
)]
struct ApiDoc;

#[actix_web::main]
async fn main() -> Result<(), Error> {
    HttpServer::new(move || {
        App::new().service(
            SwaggerUi::new("/swagger-ui/{_:.*}").url("/api-docs/openapi.json", ApiDoc::openapi()),
        )
    })
    .bind((Ipv4Addr::UNSPECIFIED, 8080))?
    .run()
    .await
}
```

The output is the following:

![image](https://github.com/juhaku/utoipa/assets/32905212/8e572393-5969-48a8-8b49-5fc4e2a41bd5)

**Comment by FancyQian** [comment]
+1 the same issue, can we fix that? or maintainer can tell me how to fix I can rasie pr.

**Comment by juhaku** [comment]
This is something that needs to be investigated a bit and fixing this should not be super hard thing.

There is a need to investigate where the title feature get's placed. I believe title feature goes for the underlying type itself and is not rendered on the array type. 

`test/schema_derive.rs` is a good place start and to add one test to test this case for the unnamed struct with title. Then `utoipa-gen/src/component.rs.` is the place where the fix should be done. https://github.com/juhaku/utoipa/blob/e0c5aa749ac548cca7fd75544293865efe5d0231/utoipa-gen/src/component.rs#L765

The same way we are popping features here, we should pop the `Title` feature as well before it is forwarded to child type. https://github.com/juhaku/utoipa/blob/e0c5aa749ac548cca7fd75544293865efe5d0231/utoipa-gen/src/component.rs#L772-L779

**Comment by FancyQian** [comment]
@juhaku tks, but the issue should on the newtype, did this project consider newtype before? I think what’s inside of newtype is not necessary, instead we need to show the inside type title of newtype.

**Comment by juhaku** [comment]
Rust new type idiom https://doc.rust-lang.org/rust-by-example/generics/new_types.html is just a wrapper for the type inside. The rust compiler will remove the wrapper during compilation and serde https://serde.rs/json.html will just serialize to the inner type directly. Utoipa will thus behave the same way. What ever is defined inside the new type will be the actual schema.
>```rust
>#[derive(ToSchema)]
>#[schema(title = "This is an integer using the new type idiom / tuple struct")]
>pub struct NewTypeInteger(Vec<i64>);
>```
With that the code above will create Array of i64 schema. Thus the title most likely should be rendered on the Array but at the moment I believe the title will actually swim down the the i64 type.

**Comment by FancyQian** [comment]
Hi @juhaku Tks, my case is not inner Vec<i64>, my case inner a new defined type
```#[test]
fn derive_unnamed_struct_type_title() {
    #[allow(unused)]
    #[derive(Default, ToSchema, Serialize)]
    #[schema(title = "My Unnamed Struct Enum")]
    enum MyUnnamedStructEnum {
        #[default]
        Enum1,
        Enum2,
    }

    let value = api_doc! {
        #[schema(default)]
        #[derive(Default)]
        #[schema(title = "My New Type Container")]
        struct MyUnnamedStruct(MyUnnamedStructEnum);
    };

    println!("{}", value);
```
```
{"allOf":[{"$ref":"#/components/schemas/MyUnnamedStructEnum"}],"default":"Enum1"}
```
I think the point is how to add Feature::Title like your said, but how to add Feature::Title as well my case, I can fix both.

**Comment by juhaku** [comment]
That's a bummer. There is no way to add title for type's that are references https://spec.openapis.org/oas/latest.html#reference-object. The spec does not have support for title on reference types. 

In order to support this scenario there is a need to change based on presence of `Title` feature the way the schema is created. What should happen is that when there is `Title` instead of creating plain `$ref` type we should create `allOf` with title and `$ref` inside. While this should work it is not official but merely a hack around the spec.

**Comment by juhaku** [comment]
There is now fix coming for the both issues in PR: #1101
